### PR TITLE
Bump cpubuilder dockerfile image to newer multi-arch version.

### PR DESCRIPTION
**Pull request by ScottTodd**
In https://github.com/iree-org/base-docker-images/pull/11, I replaced the single-architecture `cpubuilder_ubuntu_jammy_x86_64` dockerfile with a multi-architecture `cpubuilder_ubuntu_jammy` dockerfile. This checks that the `linux/amd64` platform build of the dockerfile still works for our usage.

### CODE REVIEW

**Review by marbre — APPROVED** [review]

**Comment by ScottTodd** [comment]
Oh, I forgot to merge this. At least this validated that the new multi-arch version didn't break these workflows. Now that https://github.com/iree-org/base-docker-images/pull/14 has landed, I can just bump all the dockerfile images to the latest, including the *San workflows.

**Comment by ScottTodd** [comment]
Updated all the dockerfile refs. Will wait until the builds complete to merge (or I'll forget about this again and merge it later :P)

**Pull request merged**
